promtimer/cbstats.py

```diff
@@ -429,14 +435,36 @@ def get_stats_sources(cluster, user, password):
                 if host is None:
                     host = '127.0.0.1'
                 services = node['services']
-                port = services['mgmt']
-                source = ServerNode(host, port, user, password)
+                port = services['mgmtSSL'] if secure else services['mgmt']
+                source = ServerNode(host, port, user, password, secure)
                 result.append(source)
             return result
         except OSError as err:
             logging.error('error: can\'t access cluster: {}'.format(err))
             return []
 
+    @staticmethod
+    def get_stats_sources_from_nodes(nodes, user, password):
+        result = []
+        for node in nodes:
+            secure = node.startswith("https://")
+            response = util.execute_request(node, 'nodes/self',
+                                            username=user, password=password)
+            nodes_self = json.loads(response.read())
+            real_hostname = nodes_self.get('hostname')
```

Comment:
Oh now that I read this code, I see you're getting the real name to display in the short name.

Reply:
Indeed

Reply:
Added a comment to describe what it's doing